src/stdlib/SDL_stdlib.c

```diff
@@ -504,21 +504,40 @@ int SDL_abs(int x)
 }
 
 #if defined(HAVE_CTYPE_H)
+int SDL_isalpha(int x) { return isalpha(x); }
+int SDL_isalnum(int x) { return isalnum(x); }
 int SDL_isdigit(int x) { return isdigit(x); }
+int SDL_isxdigit(int x) { return isxdigit(x); }
+int SDL_ispunct(int x) { return ispunct(x); }
 int SDL_isspace(int x) { return isspace(x); }
 int SDL_isupper(int x) { return isupper(x); }
 int SDL_islower(int x) { return islower(x); }
+int SDL_isprint(int x) { return isprint(x); }
+int SDL_isgraph(int x) { return isgraph(x); }
+int SDL_iscntrl(int x) { return iscntrl(x); }
 int SDL_toupper(int x) { return toupper(x); }
 int SDL_tolower(int x) { return tolower(x); }
 #else
+int SDL_isalpha(int x) { return (SDL_isupper(x)) || (SDL_islower(x)); }
+int SDL_isalnum(int x) { return (SDL_isalpha(x)) || (SDL_isdigit(x)); }
 int SDL_isdigit(int x) { return ((x) >= '0') && ((x) <= '9'); }
+int SDL_isxdigit(int x) { return (SDL_isalpha(x)) || (SDL_isdigit(x)); }
+int SDL_ispunct(int x) { return (SDL_isprint(x)) && (!SDL_isalnum(x)); }
```

Comment:
According to [this](http://www.cplusplus.com/reference/cctype/), `SDL_ispunct` should return false for space, however space is printable and not alphanumeric, meaning that this definition would return true. I'd suggest using `SDL_isgraph` instead of `SDL_isprint`.

Reply:
Yeah, I looked at C++ Reference too. Looks like I didn't look at it close enough (it says `isgraph` instead of `isprint`, contrary to my memory).

Reply:
I've opened #4079 to fix this.